app/controllers/streams_controller.rb

```diff
@@ -10,20 +10,29 @@ def show
     response.headers['Content-Type'] = 'text/event-stream'
     response.headers['Cache-Control'] = 'no-cache'
 
-    @job = Job.find(params[:id])
-    @execution = JobExecution.find_by_id(@job.id)
+    return response.stream.close unless job.active? && execution
 
-    streamer = EventStreamer.new(response.stream, &method(:event_handler))
-
-    return response.stream.close unless @job.active? && @execution
-
-    @execution.viewers.push(current_user)
+    execution.viewers.push(current_user)
     ActiveRecord::Base.clear_active_connections!
-    streamer.start(@execution.output)
+
+    sleep(0.1) until execution.active?
```

Comment:
Hmm, the best solution is probably to just block on the event streamer output. I didn't really want to  ouch it, but it's cleaner.